pkg/controller/schedulingpreference/scheduler.go

```diff
@@ -0,0 +1,148 @@
+/*
```

Comment:
I think this controller belongs in its own package (e.g. `pkg/controller/schedulingmanager`).

Reply:
I also think that `schedulingManager` is a better name and pkg rather then `schedulingPreference`.